src/core/xfa/layout.js

```diff
@@ -325,18 +335,27 @@ function checkDimensions(node, space) {
       // is breakable then we can return true.
       if (node.h !== "" && !node[$isSplittable]()) {
         [, , , h] = getTransformedBBox(node);
-        return Math.round(h - space.height) <= 1;
+        return Math.round(h - space.height) <= ERROR;
       }
       // Else wait and see: this node will be layed out itself
       // in the provided space and maybe a children won't fit.
+
+      if (node.w === "" || Math.round(w - space.width) <= ERROR) {
```

Comment:
@calixteman It seems that LGTM complains about this line, since `w === undefined` always holds here; which I'm guessing is probably not what you actually want?

Could we simplify all of the code in the various `case`s by replacing https://github.com/mozilla/pdf.js/blob/c33bf0b5e8dc09ed010d9c4a1dcdaa98975b6378/src/core/xfa/layout.js#L271
with unconditionally doing `const [, y, w, h] = getTransformedBBox(node);` instead?

If, so I can submit a patch to fix that!